File: forge/src/main/java/dev/obscuria/fragmentum/forge/ForgePlatform.java
package dev.obscuria.fragmentum.forge;

import dev.obscuria.fragmentum.Platform;
import net.minecraftforge.fml.ModList;
import net.minecraftforge.fml.loading.FMLEnvironment;
import net.minecraftforge.fml.loading.FMLLoader;

public final class ForgePlatform implements Platform
{
    @Override
    public String getEnvironmentName()
    {
        return "Forge";
    }

    @Override
    public boolean isModLoaded(String modId) {

        return ModList.get().isLoaded(modId);
    }

    @Override
    public boolean isDevelopmentEnvironment() {

        return !FMLLoader.isProduction();
    }

    @Override
    public boolean isClient()
    {
        return FMLEnvironment.dist.isClient();
    }

    @Override
    public boolean isDedicatedServer()
    {
        return FMLEnvironment.dist.isDedicatedServer();
    }
}

File: forge/src/main/kotlin/dev/obscuria/fragmentum/forge/platform/ForgePlatform.kt
package dev.obscuria.fragmentum.forge.platform

import dev.obscuria.fragmentum.api.Platform
import net.minecraftforge.fml.ModList
import net.minecraftforge.fml.loading.FMLEnvironment
import net.minecraftforge.fml.loading.FMLLoader

class ForgePlatform : Platform
{
    override fun getPlatformName(): String
    {
        return "Forge"
    }

    override fun isModLoaded(modId: String): Boolean
    {
        return ModList.get().isLoaded(modId)
    }

    override fun isClient(): Boolean
    {
        return FMLEnvironment.dist.isClient
    }

    override fun isDevelopmentEnvironment(): Boolean
    {
        return !FMLLoader.isProduction()
    }
}